core/platform-api/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/DefaultWorkspaceConfigValidator.java

```diff
@@ -30,7 +30,7 @@
 @Singleton
 public class DefaultWorkspaceConfigValidator implements WorkspaceConfigValidator {
     /* should contain [3, 20] characters, first and last character is letter or digit, available characters {A-Za-z0-9.-_}*/
-    private static final Pattern WS_NAME = Pattern.compile("[\\w][\\w\\.\\-]{1,18}[\\w]");
+    private static final Pattern WS_NAME = Pattern.compile("[a-zA-Z0-9][-a-zA-Z0-9.]{1,18}[a-zA-Z0-9]");
```

Comment:
You made a small mistake here as `\\w` includes underscore `_` as well, so the second [[-a-zA-Z0-9.]] should include underscore.